api/src/test/java/uk/gov/dhsc/htbhf/claimant/EmailTemplateConfigTest.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.dhsc.htbhf.claimant;
+
+import org.junit.jupiter.api.Test;
+import uk.gov.dhsc.htbhf.claimant.message.payload.EmailType;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowableOfType;
+
+class EmailTemplateConfigTest {
+
+    private static final String TEMPLATE_ID1 = "TemplateId1";
+    private static final Map<EmailType, String> TEMPLATE_IDS = Map.of(EmailType.NEW_CARD, TEMPLATE_ID1);
+    private EmailTemplateConfig emailTemplateConfig = new EmailTemplateConfig(TEMPLATE_IDS);
+
+    @Test
+    void shouldAddTemplateId() {
+        //Given
+        //When
+        String templateId = emailTemplateConfig.getTemplateIdForEmail(EmailType.NEW_CARD);
+        //Then
+        assertThat(templateId).isEqualTo(TEMPLATE_ID1);
+    }
+
+    @Test
+    void shouldFailToFindTemplateIdIfNotStored() {
+        //Given no templateIds stored
+        //When
+        IllegalArgumentException caught = catchThrowableOfType(
+                () -> emailTemplateConfig.getTemplateIdForEmail(EmailType.PAYMENT),
+                IllegalArgumentException.class);
+        //Then
+        assertThat(caught).hasMessage("No template id defined for email type: PAYMENT");
```

Comment:
This seems a bit temporary (we'll eventually have templates for all email types). Perhaps we should instead have a parameterised test with an EnumSource that confirms that all email types have a template id? This might need to be a spring boot test, so we can obtain the correct config.

Reply:
In the context of the test, this is fine as the Map of templateIds is created in the test and controlled in the test. Seems overkill to make this a Spring Boot test when its simply testing that an Exception is thrown if we request a templateId for an EmailType that isn't configured.